File: WorkoutApp/Models/Day.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.Text.Json.Serialization;

namespace WorkoutApp.Models
{
    public class Day
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? id { get; set; }

        public string workoutId { get; set; } = null!;

        public string name { get; set; } = null!;

        public int? order { get; set; } = 0;

        public string dayOfWeek { get; set; } = null!;

        public List<Set>? sets { get; set; } = new List<Set>();

    }
    public class Set
    {
        public string name { get; set; } = null!;
        public int order { get; set; } = 0;
        public List<Rep>? reps { get; set; } = new List<Rep>();
    }

    public class Rep
    {
        public int? numReps { get; set; } = null;
        public int? weight { get; set; } = null;
    }
}

File: WorkoutApp/ClientApp/src/app/classes/day.ts
export class Day {
  id?: string;
  workoutId?: string;
  name?: string;
  order?: number;
  dayOfWeek?: string;
  sets: Array<Set>;

  constructor() {
    this.workoutId = "";
    this.name = "";
    this.order = undefined;
    this.dayOfWeek = "";
    this.sets = [];
  }
}

export class Set {
  name?: string;
  order?: number;
  reps: Array<Rep>;

  constructor() {
    this.name = undefined;
    this.order = undefined;
    this.reps = [];
  }
}

export class Rep {
  numReps?: number;
  weight?: number;

  constructor() {
    this.numReps = undefined;
    this.weight = undefined;
  }
}

export class DayOfWeek {
  dayName?: string;
  dayValue?: number;

  constructor(d: string, v: number) {
    this.dayName = d;
    this.dayValue = v;
  }
}
